content/src/content/docs/learn/tutorials/first-endpoint.mdx

```diff
@@ -0,0 +1,262 @@
+      NodeHttpServer.layer(() => createServer(app), { port: 3000 }),
+    ),
+  ),
+)
+```
+
+And we can run the resulting code like so:
+
+```shell
+$ bun add @effect/platform @effect/platform-node
+$ bun index.ts
+```
+
+We haven't changed anything about how the server behaves. It still exposes a
+single endpoint, `http://localhost:3000/health`, and that still returns `ok`.
+But we've wrapped the Express.js app in a way that will allow us to define new
+endpoints using Effect while still responding to our existing endpoints in
+Express.js.
+
+There's a lot of new things happening at the bottom of our file. For the time
+being we're going to ignore most of it and focus on migrating an endpoint to
+Effect. I promise by the time we're done, you'll understand every line we added
+above.
+
+First, let's break out our router, `HttpRouter.empty`, out into its own variable
+so it's easier to add to.
+
+```ts twoslash
+import { HttpRouter } from "@effect/platform"
+// ---cut---
+const router = HttpRouter.empty
+```
+
+And let's define the function we want to run when we hit the `/health` endpoint:
+
+```ts twoslash
+import { HttpServerResponse } from "@effect/platform"
+// ---cut---
+function health() {
+  return HttpServerResponse.text("ok")
+}
+```
+
+`HttpServerResponse` is Effect's class for generating HTTP responses, we need to
+use it for any responses returned from our endpoints defined with Effect.
+
+The way we wire these things together is going to look a bit strange, but bear
+with me.
+
+```ts twoslash
+import { HttpServerResponse, HttpRouter } from "@effect/platform"
+import { Effect } from "effect"
+// ---cut---
+function health() {
+  return HttpServerResponse.text("ok")
+}
+
+const router = HttpRouter.empty.pipe(
+  HttpRouter.get("/health", Effect.sync(health)),
+)
+```
+
+What on earth are `Effect.sync` and `.pipe`?
+
+## The Effect Type
+
+At the core of Effect is... well, the `Effect`. You can think of an `Effect` as
+a lazy computation, similar to a function that hasn't been called yet.
+
+Here's an example of an `Effect` that returns the string `"Hello, world!"`:
+
+```ts twoslash
+import { Effect } from "effect"
+// ---cut---
+Effect.promise(async () => "Hello, world!")
+```
+
+`Effect.promise` takes a function that returns a `Promise` and creates an
+`Effect` out of it. `Effect.sync` works exactly the same, except it takes a
+synchronous function instead of an asynchronous one.
+
+If we want to run this Effect, we can use `Effect.runPromise`:
+
+```ts twoslash {2}
+import { Effect } from "effect"
+// ---cut---
+const promise = Effect.promise(async () => "Hello, world!")
+const result = await Effect.runPromise(promise);
+console.log(result)
+// => "Hello, world!"
+```
+
+What if you want to run an Effect that calls another Effect? You can use
+`Effect.gen` to do that:
+
+```ts twoslash {2-5}
+import { Effect } from "effect"
+// ---cut---
+const promise = Effect.promise(async () => "Hello, world!")
```

Comment:
Might be worth renaming this from `promise` to something else to avoid a user interpreting the variable as a `Promise` instead of an `Effect`.